components/org.wso2.carbon.business.rules.web/src/utils/AuthManager.jsx

```diff
@@ -115,64 +166,78 @@ export default class AuthManager {
+     * Authenticates with refresh token, when remember me has been set
+     * @returns {Promise}
      */
-    static logout() {
+    static authenticateWithRefreshToken(){
         return new Promise((resolve, reject) => {
             AuthenticationAPI
-                .logout(AuthManager.getUser().token)
-                .then(() => {
-                    AuthManager.clearUser();
-                    window.localStorage.clear();
-                    AuthManager.deleteCookie(RTK);
+                .getAccessTokenWithRefreshToken()
+                .then((response) => {
+                    const { pID, lID, validityPeriod } = response.data;
+
+                    const username = AuthManager.isRememberMeSet() ? window.localStorage.getItem('username') : AuthManager.getUser().username;
+                    AuthManager.setUser({
+                        username: username,
+                        SDID: pID,
+                        validity: validityPeriod,
+                        expires: AuthManager.calculateExpiryTime(validityPeriod),
+                    });
+                    // If rememberMe, set refresh token into a persistent cookie else session cookie.
+                    const refreshTokenValidityPeriod = AuthManager.isRememberMeSet() ? REFRESH_TOKEN_VALIDITY_PERIOD : null;
+                    AuthManager.setCookie(RTK, lID, refreshTokenValidityPeriod, window.contextPath);
                     resolve();
                 })
                 .catch(error => reject(error));
         });
     }
 
     /**
-     * Set session cookie.
+     * Logout user by revoking tokens and clearing the session.
      *
-     * @param {string} name Name of the cookie
-     * @param {string} value Value of the cookie
-     * @param {number} expiresIn Number of milliseconds to expire the cookie
+     * @returns {Promise} Promise
      */
-    static setSessionCookie(name, value, expiresIn) {
-        let expires = '';
-        if (expiresIn) {
-            const d = new Date();
-            d.setTime(d.getTime() + expiresIn);
-            expires = `expires=${d.toUTCString()};`;
-        }
-        document.cookie = `${name}=${value};${expires}path=${appContext};Secure`;
+    static logout() {
+        return new Promise((resolve, reject) => {
+            AuthenticationAPI
+                .logout(AuthManager.getUser().SDID)
+                .then(() => {
+                    AuthManager.discardSession();
+                    resolve();
+                })
+                .catch(error => reject(error));
+        });
     }
 
     /**
-     * Get session cookie by name.
-     *
-     * @param {string} name Name of the cookie
-     * @returns {string} Content
+     * Get JavaScript accessible cookies saved in browser, by giving the cooke name.
+     * @param {String} name : Name of the cookie which need to be retrived
+     * @returns {String|null} : If found a cookie with given name , return its value,Else null value is returned
```

Comment:
minor formatting. Shall we remove space before comma and add after that

Reply:
This has been already addressed. _(See https://github.com/wso2/carbon-analytics/blob/master/components/org.wso2.carbon.business.rules.web/src/utils/AuthManager.js#L301)_.